ComponentKit/Core/CKBuildComponent.mm

```diff
@@ -0,0 +1,63 @@
+//
+//  CKBuildComponent.m
+//  ComponentKit
+//
+//  Created by Oliver Rickard on 3/28/17.
+//
+//
+
+#import "CKBuildComponent.h"
+
+#import "CKComponentBoundsAnimation.h"
+#import "CKComponentBoundsAnimationPredicates.h"
+#import "CKComponentInternal.h"
+#import "CKComponentScopeRoot.h"
+#import "CKComponentSubclass.h"
+#import "CKThreadLocalComponentScope.h"
+
+static CKComponentBoundsAnimation boundsAnimationFromPreviousScopeRoot(CKComponentScopeRoot *newRoot, CKComponentScopeRoot *previousRoot)
+{
+  NSMapTable *scopeFrameTokenToOldComponent = [NSMapTable strongToStrongObjectsMapTable];
+  [previousRoot
+   enumerateComponentsMatchingPredicate:&CKComponentBoundsAnimationPredicate
+   block:^(id<CKScopedComponent> component) {
+     CKComponent *oldComponent = (CKComponent *)component;
+     id scopeFrameToken = [oldComponent scopeFrameToken];
+     if (scopeFrameToken) {
+       [scopeFrameTokenToOldComponent setObject:oldComponent forKey:scopeFrameToken];
+     }
+   }];
```

Comment:
you might want to use a std::unordered_map. if you wanted you could even use __unsafe_unretained for the keys since they are guaranteed not to go out of scope during this function's execution.